### PR TITLE
[Serializer] Document the CsvEncoder and the YamlEncoder

### PR DESCRIPTION
**Pull request by GuilhemN**
Document the CsvEncoder and the YamlEncoder introduced in https://github.com/symfony/symfony/pull/19197 and https://github.com/symfony/symfony/pull/19326.

### CODE REVIEW

**Review by Nek- — APPROVED** [review]
:+1:

File: serializer/encoders.rst [hunk 1]
```diff
@@ -56,8 +69,7 @@ Be aware that this encoder will consider keys beginning with ``@`` as attributes
     //     <foo bar="value" />
     // </response>
 
-The ``JsonEncoder``
+The ``YamlEncoder``
 ~~~~~~~~~~~~~~~~~~~
 
-The ``JsonEncoder`` is much simpler and is based on the PHP
-:phpfunction:`json_encode` and :phpfunction:`json_decode` functions.
+This encoder requires the :doc:`Yaml Component </components/yaml>`.
```
Comment (HeahDude) [inline serializer/encoders.rst]: Missing a version added note for this one.

**Review by HeahDude — CHANGES_REQUESTED** [review]

**Review by HeahDude — APPROVED** [review]
Ok I've missed it :)

**Comment by wouterj** [comment]
Thanks @GuilhemN for documenting these new serializers! Unfortunately, we've waited so long with merging this PR, we cannot merge it in 3.2 (as it's no longer maintained). So I've merged this in 3.3.

I've also slightly changed the wording in b3f570e , because it didn't read nicely anymore now you've moved the json encoder above.

**Pull request closed without merging**